src/middleware/pinataUtils.ts

```diff
@@ -2,65 +2,44 @@
  * Pinata cloud functions helper
  */
 
-import Axios, { AxiosResponse } from 'axios';
-// tslint:disable-next-line:no-var-requires
-const FormData = require('form-data');
-import fs = require('fs');
+import pinataSDK, { PinataPinResponse } from '@pinata/sdk';
 
 import ReadableStream = NodeJS.ReadableStream;
 import config from '../config';
 
-export const pinFile = (
-  file: ReadableStream,
-  filename: String = 'untitled',
-  encoding: string,
-): Promise<AxiosResponse> => {
-  const data = new FormData();
-  data.append('file', file, { filename, encoding });
+let _pinata: pinataSDK;
 
-  if (filename) {
-    const metadata = JSON.stringify({
-      name: filename,
+export const getPinata = (): pinataSDK => {
+  if (!_pinata) {
+    _pinata = new pinataSDK({
+      pinataApiKey: config.get('PINATA_API_KEY') as string,
+      pinataSecretApiKey: config.get('PINATA_SECRET_API_KEY') as string,
     });
-    data.append('pinataMetadata', metadata);
   }
+  return _pinata;
+};
 
-  return Axios.post('https://api.pinata.cloud/pinning/pinFileToIPFS', data, {
-    maxContentLength: Infinity, // this is needed to prevent Axios from throw error with large files
-    headers: {
-      'Content-Type': `multipart/form-data; boundary=${data._boundary}`,
-      pinata_api_key: config.get('PINATA_API_KEY') as string,
-      pinata_secret_api_key: config.get('PINATA_SECRET_API_KEY') as string,
-    },
+export const pinFile = (
+  file: ReadableStream,
+  filename: String = 'untitled',
```

Comment:
suggestion: lets name it something like temp_giveth_project_image so that we can debug later if needed in some edge case

Reply:
That can be set on the caller side.... it's a general function used on any kind of upload